File: python/链表/328. 奇偶链表.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def oddEvenList(self, head: ListNode) -> ListNode:
        if not head: return None
        odd,even = head,head.next
        while even and even.next:
            odd.next = even.next
            even.next = even.next.next
            odd = odd.next
            even = even.next
        odd.next = even
        return head

File: go/link/328. 奇偶链表.go
package link

func oddEvenList(head *ListNode) *ListNode {
	if head == nil {
		return nil
	}
	pre := head.Next
	odd, even := head, head.Next
	for even != nil && even.Next != nil {
		odd.Next = even.Next
		even.Next = even.Next.Next
		odd = odd.Next
		even = even.Next
	}
	odd.Next = pre
	return head
}
